jme3-core/src/main/java/com/jme3/animation/SkeletonControl.java

```diff
@@ -443,9 +459,19 @@ public Node getAttachmentsNode(String boneName) {
                     + "in the skeleton.");
         }
 
-        Node n = b.getAttachmentsNode();
-        Node model = (Node) spatial;
-        model.attachChild(n);
+        updateTargetsAndMaterials(spatial);
+        Node n = b.getAttachmentsNode(targetGeometry);
```

Comment:
You should pass the entire list of geoms and let the Bone class find the one that it affects.